src/UartSerial.cpp

```diff
@@ -130,10 +136,43 @@ int UartSerial::read()
 
 void UartSerial::write(uint8_t byte_)
 {
-  ::write(_fd, &byte_, 1);
+  if(_fd != -1)
```

Comment:
Would like to see this as an early return to be consistent with your other functions. Looks like a maintenance headache.